### PR TITLE
chore: derive compactors DF memory pool size from a percent of total 

### PR DESCRIPTION
**Pull request by Joe-Blount**
This PR automates maintenance of the compactor's memory given to DF jobs.

The amount of memory going to the compactor's DF jobs (`INFLUXDB_IOX_EXEC_MEM_POOL_BYTES`) should be a sizable fraction of the memory available to the compactor. This causes a maintenance challenge with `INFLUXDB_IOX_EXEC_MEM_POOL_BYTES` needing changed each time the compactor's spec is changed.  Additionally, in CST the compactor's memory spec is derived at render time, which complicates deriving values from it.

This PR adds `INFLUXDB_IOX_EXEC_MEM_POOL_PERCENT`, which sets `INFLUXDB_IOX_EXEC_MEM_POOL_BYTES` to the specified percent of the detected cgroup memory limit.  If the percent is extreme (<20 or >90), or if we fail to read or parse the cgroup memory limit, or if the cgroup memory limit is suspect (<1GB  or >512 GB), we fall back to `INFLUXDB_IOX_EXEC_MEM_POOL_BYTES` and log a warning explaining why.

The default for `INFLUXDB_IOX_EXEC_MEM_POOL_BYTES` is changed from 8 to 16 GB in this PR, which is the default set in k8s.

If this PR is approved, the deployment sequence will be:
1) add `INFLUXDB_IOX_EXEC_MEM_POOL_PERCENT` to k8s config, let it deploy
2) merge this PR
3) then consider removing the setting of `INFLUXDB_IOX_EXEC_MEM_POOL_BYTES` in k8s, letting it stick to the default (16GB now), to avoid confusion from having two values set in k8s?

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: clap_blocks/src/compactor.rs [hunk 1]
```diff
@@ -73,11 +73,30 @@ pub struct CompactorConfig {
...
         action
     )]
     pub exec_mem_pool_bytes: MemorySize,
 
+    /// Overrides INFLUXDB_IOX_EXEC_MEM_POOL_BYTES to set the size of memory pool
+    /// used during compaction DF plan execution.  This value is expressed as a percent
+    /// of the memory limit for the cgroup (e.g. 70 = 70% of the cgroup memory limit).
+    /// This is converted to a byte limit as the compactor starts.
+    ///
+    /// Values below 25% or above 90% are adjusted to 20 and 90 respectively.
```
Comment (philjb) [inline clap_blocks/src/compactor.rs:86]: typo? 25 v 20

File: ioxd_compactor/src/lib.rs [hunk 2]
```diff
@@ -144,6 +146,50 @@ impl HttpApiErrorSource for IoxHttpError {
...
+    let pod_limit_str = fs::read_to_string("/sys/fs/cgroup/memory/memory.limit_in_bytes");
+    if pod_limit_str.is_ok() {
+        let pod_limit_bytes = pod_limit_str.unwrap().trim().parse::<i64>();
+        if pod_limit_bytes.is_ok() {
+            let pod_limit_bytes = pod_limit_bytes.unwrap();
+            if pod_limit_bytes >= 1024 * 1024 * 1024 && pod_limit_bytes <= 512 * 1024 * 1024 * 1024
+            {
+                let pod_df_limit = (pod_limit_bytes as f64 * percent as f64 / 100.0) as usize;
+
+                if percent >= 20 && percent <= 90 {
```
Comment (philjb) [inline ioxd_compactor/src/lib.rs:165]: did you want to force to 20% or 90% like you mentioned in the comment above? This just ignores/warns a bad percentage.

**Review by philjb — APPROVED** [review]
❤️  two comments tho you'll want to double check

File: clap_blocks/src/compactor.rs (hunk 1 above)
Comment (Joe-Blount) [reply clap_blocks/src/compactor.rs]: yeah, I changed my mind and then didn't catch all the old references...

File: clap_blocks/src/compactor.rs [hunk 3]
```diff
@@ -73,11 +73,30 @@ pub struct CompactorConfig {
...
     pub exec_mem_pool_bytes: MemorySize,
 
+    /// Overrides INFLUXDB_IOX_EXEC_MEM_POOL_BYTES to set the size of memory pool
+    /// used during compaction DF plan execution.  This value is expressed as a percent
+    /// of the memory limit for the cgroup (e.g. 70 = 70% of the cgroup memory limit).
+    /// This is converted to a byte limit as the compactor starts.
+    ///
+    /// Extreme values (<20% or >90%) are ignored and INFLUXDB_IOX_EXEC_MEM_POOL_BYTES
+    /// is used.  It will also use INFLUXDB_IOX_EXEC_MEM_POOL_BYTES if we fail to read
+    /// the cgroup limit, or it doesn't parese to a sane value.
```
Comment (philjb) [inline clap_blocks/src/compactor.rs:88]: typo: parse

**Review by philjb — APPROVED** [review]

**Pull request merged**